llama_index/query_engine/retriever_query_engine.py

```diff
@@ -48,6 +49,11 @@ def __init__(
 
         super().__init__(callback_manager)
 
+    def _get_prompts_and_modules(self) -> Tuple[PromptDictType, PromptMixinType]:
```

Comment:
would it be easier if we just return the prompts for this module and its children here, rather than doing the recursive traversal in the mixin class?

Reply:
is the idea that this function would also get/set response synthesizer prompts? 

would def make the mixin class simpler, but might lead to duplicate code if the synthesizer is used in more places